ui/store/actions.ts

```diff
@@ -5152,6 +5152,39 @@ export function setName(
   }) as any;
 }
 
+/**
+ * To create a data deletion regulation for MetaMetrics data deletion
+ */
+export function createMetaMetricsDataDeletionTask(): Promise<void> {
+  return async () => {
+    await submitRequestToBackground<void>('createMetaMetricsDataDeletionTask');
+  };
+}
```

Comment:
If this function returns a promise, shouldn't it be async?

```suggestion
export async function createMetaMetricsDataDeletionTask(): Promise<void> {
  return async () => {
    await submitRequestToBackground<void>('createMetaMetricsDataDeletionTask');
  };
}
```

I'm not sure I understand the purpose of wrapping the return statement in a callback. Is there a reason we can't do one of the following?

```suggestion
export function createMetaMetricsDataDeletionTask(): void {
  return await submitRequestToBackground<void>('createMetaMetricsDataDeletionTask');
}
```

```suggestion
export async function createMetaMetricsDataDeletionTask(): Promise<void> {
  return submitRequestToBackground<void>('createMetaMetricsDataDeletionTask');
}
```

The same applies to `updateDataDeletionTaskStatus` and `setHasMetaMetricsDataRecorded`.